shoop/utils/enums.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+from numbers import Number
+
+import six
+
+RAISE = object()
+
+
+def map_enum(enum_cls, value, default=RAISE, allow_magic_numbers=False):
+    """
+    Attempt to cast the given `value` into a value of the given `enum_cls`.
+
+    Tries by value first, then case-insensitively by name.
+
+    If casts fail, the `default` is returned (unless it is the default sentinel
+    object `RAISE`, in which case a `ValueError` is raised).
+
+    This is useful for places where it doesn't make sense to export the actual
+    enumeration classes (such as into templates, etc.) while still avoiding
+    "magic constants" (i.e. the actual enum values).  For the aforementioned
+    reason of being used in templates, it is purposefully slightly more
+    lenient in its parsing than more "core" utilities might be.
+
+    :param enum_cls: Enum class
+    :type enum_cls: enum.Enum
+    :param value: value or name
+    :type value: object|str
+    :param default: Default value to return in case of trouble
+    :type default:
+    :param allow_magic_numbers: Allow mapping via "magic numbers". If this is False and
+                                the value is a number, the default `enum_cls(value)` logic
+                                is not attempted.
```

Comment:
This allow_magic_numbers stuff does not belong to same function IMHO. A separate function would be much clearer, since it's a separate functionality. `map_enum(MyEnum, 42, allow_magic_numbers=True)` vs `map_enum_int(MyEnum, 42)`

Having the int mapping in separate function would also make this one much clearer, since the value would always be string.